vehicle/OVMS.V3/components/vehicle_fiatedoblo/src/vehicle_fiatedoblo.cpp

```diff
@@ -0,0 +1,557 @@
+      */
+
+      int changed = 0;
+      for (int i = 0; i < 8; i++){
+	if(last_data[i] != d[i]){
+	  last_data[i] = d[i];
+	  changed = 1;
+	}
+      }
+      if(changed){
+	ESP_LOGI(TAG, "changes in 0x%4x: 0x%2x 0x%2x 0x%2x 0x%2x  0x%2x 0x%2x 0x%2x 0x%2x", p_frame->MsgID, d[0], d[1], d[2], d[3], d[4], d[5], d[6], d[7]); // TODO remove
+      }
+    }
+    break;
+  case 0x5c3:
+    {
+      // Byte 2 könnte Ladestrom in bits 0011 1100 sein.
+    break;
+    }
+  }
+}
+
+void OvmsVehicleFiatEDoblo::IncomingPollReply(const OvmsPoller::poll_job_t &job, uint8_t* data, uint8_t length)
+{
+  switch (job.moduleid_rec) {
+  case 0x682:
+    // temperatures and other data from VCU
+    IncomingVCUPoll(job, data, length);
+    break;
+  case 0x694:
+    // battery details
+    IncomingBatteryPoll(job, data, length);
+    break;
+  case 0x7e8:
+    // more details
+    IncomingVINPoll(job, data, length);
+    break;
+  default:
+    ESP_LOGI(TAG, "unexpected poll reply on ID 0x%4x (pid = 0x%4x, len = %d)", job.moduleid_rec, job.pid, length);
+    ESP_LOGI(TAG, "unexpected poll reply 0x%4x (pid = 0x%x, len = %d): 0x%2x %c 0x%2x %c 0x%2x %c 0x%2x %c  0x%2x %c 0x%2x %c 0x%2x %c 0x%2x %c",
+	     job.moduleid_rec, job.pid, length,
+	     data[0], (char)data[0],
+	     data[1], (char)data[1],
+	     data[2], (char)data[2],
+	     data[3], (char)data[3],
+	     data[4], (char)data[4],
+	     data[5], (char)data[5],
+	     data[6], (char)data[6],
+	     data[7], (char)data[7]);
+  }
+  /*    case 0x1f:  // runtime since engine start
+      ESP_LOGI(TAG, "runtime since engine start %d", value2);
+      StandardMetrics.ms_v_env_drivetime->SetValue(value2);
+      break;
+    case 0x0d:  // Speed
+      StandardMetrics.ms_v_pos_speed->SetValue(value1);
+      break;
+  */
+}
+
+/**
+ * Handle incoming messages from VIN-poll
+ */
+void OvmsVehicleFiatEDoblo::IncomingVINPoll(const OvmsPoller::poll_job_t &job, uint8_t* data, uint8_t length)
+{
+  if(job.pid == 0x02) {
+    if(job.mlremain > 13) {
+      vin.clear();
+    }
+    vin.append((char*)data, length);
+    if(job.mlremain == 0) {
+      ESP_LOGI(TAG, "rrrrreceived VIN: %s", vin.c_str());
+      StdMetrics.ms_v_vin->SetValue(vin);
+    }
+  }else{
+    ESP_LOGI(TAG, "unexpected poll reply 0x%4x (pid = 0x%x, len = %d): 0x%2x %c 0x%2x %c 0x%2x %c 0x%2x %c  0x%2x %c 0x%2x %c 0x%2x %c 0x%2x %c",
+	     job.moduleid_rec, job.pid, length,
+	     data[0], (char)data[0],
+	     data[1], (char)data[1],
+	     data[2], (char)data[2],
+	     data[3], (char)data[3],
+	     data[4], (char)data[4],
+	     data[5], (char)data[5],
+	     data[6], (char)data[6],
+	     data[7], (char)data[7]);
+  }
+  // TODO  ESP_LOGI(TAG, "mlremain %d", job.mlremain);
+}
+
+/**
+ * Handle incoming messages from Battery-poll
+ *
+ */
+void OvmsVehicleFiatEDoblo::IncomingBatteryPoll(const OvmsPoller::poll_job_t &job, uint8_t* data, uint8_t length)
+{
+  int value1 = (int)data[0];
+  int value2 = ((int)data[0] << 8) + (int)data[1];
```

Comment:
This is most probably wrong: if data[0] + data[1] build a signed 16 bit value, you shouldn't cast to int before building the full 16 bit value.